### ISSUE TITLE
HTTPS redirect not working

### ISSUE THREAD

**Issue by hex-m** [certbot/certbot#3742]
I have Apache running on CentOS 7 with the default configuration of phpMyAdmin. So basically `yum install epel-release python-certbot-apache httpd mod_ssl phpmyadmin` without special configuration.
certbot --apache does run fine in "easy" mode but when I try "secure" (Make all requests redirect to secure HTTPS access) I get errors and the redirect does not work.
Terminal output:
```
Error while running apachectl configtest.

AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using fe80::3465:XXX. Set the 'ServerName' directive globally to suppress this message
(2)No such file or directory: AH02291: Cannot access directory '/var/log/apache2/' for error log of vhost defined at /etc/httpd/conf.d/le-redirect-servername.example.com.conf:1
AH00014: Configuration check failed

IMPORTANT NOTES:
 - We were unable to restart web server
```
The mentioned path /var/log/apache2/ does of course not exist on a CentOS System. Maybe that's part of the problem?
/var/log/httpd/error_log:
```
[Fri Nov 04 15:31:58.214571 2016] [mpm_prefork:notice] [pid 10052] AH00171: Graceful restart requested, doing restart
AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using fe80::3465:XXX. Set the 'ServerName' directive globally to suppress this message
[Fri Nov 04 15:31:58.262104 2016] [auth_digest:notice] [pid 10052] AH01757: generating secret for digest authentication ...
[Fri Nov 04 15:31:58.262761 2016] [lbmethod_heartbeat:notice] [pid 10052] AH02282: No slotmem from mod_heartmonitor
[Fri Nov 04 15:31:58.278655 2016] [mpm_prefork:notice] [pid 10052] AH00163: Apache/2.4.6 (CentOS) OpenSSL/1.0.1e-fips PHP/5.4.16 configured -- resuming normal operations
[Fri Nov 04 15:31:58.278664 2016] [core:notice] [pid 10052] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND'
```

**Comment by pde** [comment]
@joohoi thoughts?

**Comment by pde** [comment]
@hex-m can you post your configuration file?

**Comment by joohoi** [comment]
My guess would be that mod_rewrite isn't enabled, and we're not handling module activation in CentOS currently as the mechanism there is pretty stoic. @hex-m could you check your configuration, if it has 'LoadModule rewrite_module modules/mod_rewite.so' commented out somewhere.

**Comment by hex-m** [comment]
The mentioned config line is in /etc/httpd/conf.modules.d/00-base.conf so I guess that's not the problem. This is my /etc/httpd/conf.d/ssl.conf which got created by `yum install mod_ssl` and modified by certbot.
```
Listen 443 https
SSLPassPhraseDialog exec:/usr/libexec/httpd-ssl-pass-dialog
SSLSessionCache         shmcb:/run/httpd/sslcache(512000)
SSLSessionCacheTimeout  300
SSLRandomSeed startup file:/dev/urandom  256
SSLRandomSeed connect builtin
SSLCryptoDevice builtin

<VirtualHost _default_:443>
ErrorLog logs/ssl_error_log
TransferLog logs/ssl_access_log
LogLevel warn
SSLEngine on
SSLProtocol all -SSLv2
SSLCipherSuite HIGH:MEDIUM:!aNULL:!MD5:!SEED:!IDEA
SSLCertificateFile /etc/letsencrypt/live/server.example.com/cert.pem
SSLCertificateKeyFile /etc/letsencrypt/live/server.example.com/privkey.pem

<Files ~ "\.(cgi|shtml|phtml|php3?)$">
    SSLOptions +StdEnvVars
</Files>

<Directory "/var/www/cgi-bin">
    SSLOptions +StdEnvVars
</Directory>

BrowserMatch "MSIE [2-5]" \
         nokeepalive ssl-unclean-shutdown \
         downgrade-1.0 force-response-1.0

CustomLog logs/ssl_request_log \
          "%t %h %{SSL_PROTOCOL}x %{SSL_CIPHER}x \"%r\" %b"

ServerName server.example.com
SSLCertificateChainFile /etc/letsencrypt/live/server.example.com/chain.pem
</VirtualHost>
```

**Comment by joohoi** [comment]
@hex-m is the mod_rewrite line commented out? If it is, please remove the comment, thus activating it. The "secure" - setting in apache plugin is trying to use mod_rewrite to redirect user from HTTP to HTTPS

**Comment by hex-m** [comment]
@joohoi the line is not commented out. Here is my httpd.conf.
```
ServerRoot "/etc/httpd"
Listen 80
Include conf.modules.d/*.conf
User apache
Group apache
ServerAdmin admin@example.com
<Directory />
    AllowOverride none
    Require all denied
</Directory>
<Directory "/var/www/html">
    Options Indexes FollowSymLinks
    AllowOverride None
    Require all granted
</Directory>

<IfModule dir_module>
    DirectoryIndex index.html
</IfModule>

<Files ".ht*">
    Require all denied
</Files>

ErrorLog "logs/error_log"
LogLevel warn

<IfModule log_config_module>
    LogFormat "%h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\"" combined
    LogFormat "%h %l %u %t \"%r\" %>s %b" common
    <IfModule logio_module>
      LogFormat "%h %l %u %t \"%r\" %>s %b \"%{Referer}i\" \"%{User-Agent}i\" %I %O" combinedio
    </IfModule>
    CustomLog "logs/access_log" combined
</IfModule>

<IfModule alias_module>
    ScriptAlias /cgi-bin/ "/var/www/cgi-bin/"
</IfModule>

<Directory "/var/www/cgi-bin">
    AllowOverride None
    Options None
    Require all granted
</Directory>

<IfModule mime_module>
    TypesConfig /etc/mime.types
    AddType application/x-compress .Z
    AddType application/x-gzip .gz .tgz

    AddType text/html .shtml
    AddOutputFilter INCLUDES .shtml
</IfModule>
EnableSendfile on
IncludeOptional conf.d/*.conf
```

**Comment by joohoi** [comment]
Oh, I think you are running to the problem that was recently fixed: ( #3448 ) but certbot version installed from epel is not that recent.

**Comment by hex-m** [comment]
That makes sense. Thank you!
I just checked and yes, epel-release currently has version 0.8.1 from July 2016 and the bug was fixed in August.
